src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1077,10 +1074,14 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
                         {
                             if (len > ReadBuffer.Size)
                             {
+                                var oversizeBuffer = ReadBuffer.AllocateOversize(len);
+
```

Comment:
In this case, it would look something like this:
_'Dispose `ReadBuffer` and after that call `ReadBuffer.AllocateOversize`'_
And while it would work (as we are only disposing internal `CancellationTokenSource`), I do not think we should do so, as any change in `ReadBuffer.Dispose` could break this behavior.
For now, I've reversed this change and added commentary.

Reply:
I agree.
This was indeed an oversight on my part.
Anyway, since `ReadBuffer` isn't really a fully implemented IDisposable (not throwing `ObjectDisposedException` on public methods) and I think we can allow ourselves to leak the implementation out of the abstraction here.
On the other hand I'm also fine with reverting https://github.com/npgsql/npgsql/pull/2911/commits/54683e6345708e8dfc7892124cfcb671290d63a3 because your initial implementation was correct and sane.